hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/FrameRunner.java

```diff
@@ -101,11 +103,23 @@ public HederaEvmTransactionResult runToCompletion(
 
         // And return the result, success or failure
         final var gasUsed = effectiveGasUsed(gasLimit, frame);
+        Long signerNonce = null;
+        if (hydratedEthTxData != null && sender != null) {
```

Comment:
All these changes are confusing imo:
 - The `sender != null` condition is dead code---i.e., it can never be true because of the `requireNonNull(sender)` on L#87. 
 - Adding `HydratedEthTxData` to the argument list is misleading because the method doesn't actually use it for anything other than deciding to set `signerNonce` in its return value.
 - **But** even that is confusing, because the client code (i.e., the code that calls this method) already knows both (1) whether `HydratedEthTxData` is non-null and (2) what the `signerNonce` is. So I'd expect the client code to either just pass in the `signerNonce` to use; or even better, for this method to know nothing about nonces at all, since the client code can already manage it.